poly/joint_test.go

```diff
@@ -0,0 +1,99 @@
+package poly
+
+import (
+	"fmt"
+	"testing"
+)
+
+/////// TESTING ///////
+
+func TestReceiverAddDealer(t *testing.T) {
```

Comment:
Tests specifically related to a particular type, e.g., Dealer, should be called 'TestDealer...' with the type name first after Test, so that we can 'go test -run Dealer*' to run only the tests related to Dealer.